src/aleph/http.clj

```diff
@@ -21,45 +21,69 @@
+   | `continue-executor`               | Optional `java.util.concurrent.Executor` which is used to handle requests passed to :continue-handler. To avoid this indirection you may specify `:none`, but in this case extreme care must be taken to avoid blocking operations on the handler's thread. |
+   | `shutdown-timeout`                | Interval in seconds within which in-flight requests must be processed, defaults to 15 seconds. A value of `0` bypasses waiting entirely. |
+
+   HTTP/1-specific options
+   | Param key                         | Description |
+   | ---                               | --- |
+   | `allow-duplicate-content-lengths` | If `true`, allows duplicate `Content-Length` headers, defaults to true. Always true for HTTP/2 |
+   | `max-initial-line-length`         | The maximum characters that can be in the initial line of the request, defaults to `8192` |
+   | `max-header-size`                 | The maximum characters that can be in a single header entry of a request, defaults to `8192` |
+   | `max-chunk-size`                  | The maximum characters that can be in a single chunk of a streamed request, defaults to `16384` |
+   | `initial-buffer-size`             | The initial buffer size of characters when decoding the request, defaults to `128` |
+   | `pipeline-transform`              | (DEPRECATED: Use `:http1-pipeline-transform` instead.) A function that takes an `io.netty.channel.ChannelPipeline` object, which represents a connection, and modifies it. |
+   | `http1-pipeline-transform`        | A function that takes an `io.netty.channel.ChannelPipeline` object, which represents an HTTP/1 connection, and modifies it. Contains the user handler at \"request-handler\". |
+   | `compression-level`               | (DEPRECATED: Use `compression-options` or leave unset.) Optional gzip/deflate compression level, `1` yields the fastest compression and `9` yields the best compression, defaults to `6` . When set, ONLY older gzip and deflate codecs are available. Overridden by `compression-options`. |
+
+   HTTP/2-specific options
+   | Param key                         | Description |
+   | ---                               | --- |
+   | `use-h2c?`                        | If `true`, uses HTTP/2 for insecure servers. Has no effect on secure servers, and upgrades are not allowed. Defaults to false. |
+   | `conn-go-away-handler`            | A connection-level cleanup handler for when a GOAWAY is received. Indicates the peer has, or soon will, close the TCP connection. Contains the last-processed stream ID.|
+   | `stream-go-away-handler`          | A stream-level cleanup handler called for streams above the last-stream-id in a GOAWAY frame. Indicates the stream will not be processed. Called with the context and the Http2GoAwayFrame. |
+   | `reset-stream-handler`            | A stream-level cleanup handler called for streams that have been sent RST_STREAM. Called with the context and the Http2ResetFrame. |
```

Comment:
Well, we talked about allowing people lower-level access. This is currently the best way that can happen with the netty multiplex code. 

Otherwise, those H2 frames will never be seen by a ring handler. They will just result in the closing of the connection or stream.

Reply:
I remember those discussions, but I can't imagine what consumers are going to do about it. (logging only?)

Reply:
Logging is a possibility, as is coordinating across streams, or any kind of general clean up.

Silly example: if someone disconnects from a multi-user chat, you could use the go-away handler to broadcast a final "goodbye" message.

Reply:
Ok, it was just out-of-curiosity.